src/vec_int16_ppc.h

```diff
@@ -0,0 +1,392 @@
+#endif
+  result = (vui16_t) vec_perm ((vui8_t) vra, (vui8_t) vra, (vui8_t) vconstp);
+#endif
+
+  return (result);
+}
+
+/** \brief Vector Shift left Halfword Immediate.
+ *
+ *	Vector Shift left Halfwords each element [0-7], 0-15 bits,
+ *	as specified by an immediate value.
+ *	The shift amount is a const unsigned int in the range 0-15.
+ *	A shift count of 0 returns the original value of vra.
+ *	Shift counts greater then 15 bits return zero.
+ *
+ *	@param vra a 128-bit vector treated as a vector unsigned short.
+ *	@param shb Shift amount in the range 0-15.
+ *	@return 128-bit vector unsigned short, shifted left shb bits.
+ */
+static inline vui16_t
+vec_slhi (vui16_t vra, const unsigned int shb)
+{
+  vui16_t lshift;
+  vui16_t result;
+
+  if (shb < 16)
+    {
+      /* Load the shift const in a vector.  The element shifts require
+         a shift amount for each element. For the immediate form the
+         shift constant is splatted to all elements of the
+         shift control.  */
+      if (__builtin_constant_p(shb))
+	lshift = (vui16_t) vec_splat_s16(shb);
+      else
+	lshift = vec_splats ((unsigned short) shb);
+
+      /* Vector Shift right bytes based on the lower 4-bits of
+         corresponding element of lshift.  */
+      result = vec_vslh (vra, lshift);
+    }
+  else
+    { /* shifts greater then 15 bits return zeros.  */
+      result = vec_xor ((vui16_t) vra, (vui16_t) vra);
+    }
+
+  return (vui16_t) result;
+}
+
+/** \brief Vector Shift Right Halfword Immediate.
+ *
+ *	Vector Shift right Halfwords each element [0-7], 0-15 bits,
+ *	as specified by an immediate value.
+ *	The shift amount is a const unsigned int in the range 0-15.
+ *	A shift count of 0 returns the original value of vra.
+ *	Shift counts greater then 15 bits return zero.
+ *
+ *	@param vra a 128-bit vector treated as a vector unsigned short.
+ *	@param shb Shift amount in the range 0-15.
+ *	@return 128-bit vector unsigned short, shifted right shb bits.
+ */
+static inline vui16_t
+vec_srhi (vui16_t vra, const unsigned int shb)
+{
+  vui16_t lshift;
+  vui16_t result;
+
+  if (shb < 16)
+    {
+      /* Load the shift const in a vector.  The element shifts require
+         a shift amount for each element. For the immediate form the
+         shift constant is splatted to all elements of the
+         shift control.  */
+      if (__builtin_constant_p(shb))
+	lshift = (vui16_t) vec_splat_s16(shb);
+      else
+	lshift = vec_splats ((unsigned short) shb);
+
+      /* Vector Shift right bytes based on the lower 4-bits of
+         corresponding element of lshift.  */
+      result = vec_vsrh (vra, lshift);
+    }
+  else
+    { /* shifts greater then 15 bits return zeros.  */
+      result = vec_xor ((vui16_t) vra, (vui16_t) vra);
+    }
+  return (vui16_t) result;
+}
+
+/** \brief Vector Shift Right Algebraic Halfword Immediate.
+ *
+ *  Vector Shift Right Algebraic Halfwords each element [0-7],
```

Comment:
capitalization again. I know it's getting old, just looking for consistency. If you don't care, we can let it go.